src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -38,6 +38,20 @@ public ServiceAuthorizationBehavior()
             _principalPermissionMode = DefaultPrincipalPermissionMode;
         }
 
+        private ServiceAuthorizationBehavior(ServiceAuthorizationBehavior other)
```

Comment:
I really like this approach better than my original idea. You don't need to do anything different on the DI injection piece with this change, you can leave it as a simple singleton as the service description is cloning it.

Reply:
The idea is the configuration one would apply when pulling the service from DI is a kind of template that allow for further override when using `ServiceBuilder.ConfigureServiceHost`.

Reply:
You would still get that with a straight Singleton. You pull it from DI and customize it, then when populating the description, it would get cloned. Then in ConfigureServiceHostBase, you would pull it from the behaviors and modify it, which would affect other instances as it's cloned.

Reply:
@mconnew then maybe that's better to drop the factory introduced by this PR. What do you think?

Reply:
@mconnew Done! Thanks